flyteplugins/go/tasks/pluginmachinery/tasklog/template.go

```diff
@@ -10,9 +10,17 @@ import (
 )
 
 func MustCreateRegex(varName string) *regexp.Regexp {
+	// TODO: Is this regex correct? Why are we allowing $? Shouldn't it be {{\s*\.%s\s*}}?
+	// Also, case insensitive matching is enabled. Is this correct?
 	return regexp.MustCompile(fmt.Sprintf(`(?i){{\s*[\.$]%s\s*}}`, varName))
 }
 
+var dynamicLogRegex = regexp.MustCompile(`(?i){{\s*.taskConfig[\.$]([a-zA-Z_]+)\s*}}`)
```

Comment:
nit: Name it more specifically... something like `taskConfigVarRegex` or something...